apps/dotcms-ui/src/app/view/components/dot-navigation/services/dot-navigation.service.ts

```diff
@@ -373,4 +383,17 @@ export class DotNavigationService {
     private setMenu(menu: DotMenu[]) {
         this._items$.next(this.addMenuLinks(menu));
     }
+
+    private getPageCurrentTitle(url: string, menu: DotMenu[]): string {
+        let title = '';
+        for (let i = 0; i < menu.length; i++) {
+            for (let j = 0; j < menu[i].menuItems.length; j++) {
+                title =
+                    url.indexOf(menu[i].menuItems[j].menuLink) >= 0
+                        ? menu[i].menuItems[j].label
+                        : title;
+            }
+        }
+        return title;
```

Comment:
Isn't easier to join all the children in one plain array and then look for the URL to match?

Reply:
done!